.github/workflows/build.yml

```diff
@@ -0,0 +1,180 @@
+    - name: Verify
+      run: make verify
+
+  test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_HOSTNAME: "localhost:5000"
+      DOCKER_NAMESPACE: m4d-system
+    steps:
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.13
+    - uses: actions/checkout@v2
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.m2/repository
+          ~/go/pkg/mod
+          ${{ github.workspace }}/hack/tools/bin
+        key: ${{ runner.os }}-ut-m2-${{ hashFiles('**/pom.xml') }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: ${{ runner.os }}-ut-m2
+    - name: Test
+      run: make test
+
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_HOSTNAME: "localhost:5000"
+      DOCKER_NAMESPACE: m4d-system
+    steps:
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.13
+    - uses: actions/checkout@v2
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.m2/repository
+          ~/go/pkg/mod
+          ${{ github.workspace }}/hack/tools/bin
+        key: ${{ runner.os }}-it-m2-${{ hashFiles('**/pom.xml') }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: ${{ runner.os }}-it-m2
+    - name: Install tools
+      run: make install-tools
+    - name: Integration tests
+      run: make run-integration-tests
+    - run: docker images
+    - name: Save images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      run: make save-images
+    - name: Upload images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: images
+        path: images.tar
+        retention-days: 1
+
+  push_images:
+    name: Push images
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_HOSTNAME: "localhost:5000"
+      DOCKER_NAMESPACE: m4d-system
+    needs: [verify, test, integration-tests]
+    if: ${{ github.event_name != 'pull_request' }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Download artifact
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: actions/download-artifact@v2
+      with:
+        name: images
+    - name: Load images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      run: docker load -i images.tar
+    - run: docker images
+    - name: Publish images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      run: sudo chmod o+x /etc/docker && make docker-retag-and-push-public && make helm-push-public
```

Comment:
helm-push-public relies on TRAVIS_TAG environment variable

Reply:
Github actions has `github.ref` but it works a bit different as it has the branch name or tag name. So we will have to redo this part. I'd propose to tackle this in a separate PR. There are also actions for publishing images. Maybe one of them already does something that we want to happen on a release vs main branch build.

Reply:
We can do a quick and dirty workaround to set TRAVIS_TAG like what's described in https://github.community/t/how-to-get-just-the-tag-name/16241/10 without doing major refactoring just to get this working because we want to soon create a tagged release of the project (I think the middle of this month is a target but unsure). Unless you think we can solve it in a separate PR in time.